contracts/src/v0.8/keeper2_0/KeeperRegistryBase2_0.sol

```diff
@@ -444,6 +446,22 @@ abstract contract KeeperRegistryBase2_0 is ConfirmedOwner, ExecutionPrevention {
     return transmitter.balance;
   }
 
+  function _blockNum() internal view returns (uint256) {
+    if (i_mode == Mode.ARBITRUM) {
+      return ARB_SYS.arbBlockNumber();
+    } else {
+      return block.number;
+    }
+  }
+
+  function _blockHash(uint256 blockNum) internal view returns (bytes32) {
+    if (i_mode == Mode.ARBITRUM) {
+      return ARB_SYS.arbBlockHash(blockNum);
```

Comment:
doesn't solidity do the same?

Reply:
oh, it returns zero [LINK](https://docs.soliditylang.org/en/v0.8.18/units-and-global-variables.html#block-and-transaction-properties)

Reply:
We've now changed it to avoid the call entirely if the block number is too old: https://github.com/smartcontractkit/chainlink/pull/8378/files#diff-a652d91a0b0fa2bc99eb95cdd8a7b1204490fe3ad9275f96aaa3422a2a064f37R23